aiogram_dialog/context/intent_filter.py

```diff
@@ -45,9 +45,34 @@ def __init__(self, storage: BaseStorage, state_groups: Dict[str, Type[StatesGrou
         self.storage = storage
         self.state_groups = state_groups
 
+    async def __call__(
+            self,
+            handler: Callable[[Union[Update, DialogUpdateEvent], Dict[str, Any]], Awaitable[Any]],
+            event: Update,
+            data: Dict[str, Any],
+    ) -> Any:
+
+        try:
+            if isinstance(event, DialogUpdateEvent):
+                await self.on_pre_process_aiogd_update(event, data)
+            elif event.message:
+                await self.on_pre_process_message(event.message, data)
+            elif event.chat_member:
+                await self.on_pre_process_message(event.message, data)
+            elif event.callback_query:
+                await self.on_pre_process_callback_query(event.callback_query, data)
+        except CancelHandler:
+            return
```

Comment:
Are you sure `CancelHandler` should be ignored?

Reply:
This is outer (pre_update) middleware, 
we just skip update,
so maybe just use other Exception for this case?